### PR TITLE
[feature] switched releasing via mvn from bintray to github-releases

### PR DESCRIPTION
**Pull request by windauer**
Replaces bintray mvn plugin with github-release-plugin by de.jutzig

### Description:

To create a draft release call: `mvn -DskipTests -Dgithub.draft=true github-release:release` in $EXIST/exist-distribution or $EXIST/exist-installer. 

The plugin is configured to be executed in the maven "deploy" phase which will be process during `mvn release:perform`.

Documentation has been added to https://github.com/eXist-db/exist/blob/develop/exist-versioning-release.md (see changed files). 

### Reference:
* https://github.com/jutzig/github-release-plugin

### Type of tests 
* none

### Known Issues
Missing tests. The plugin has been tested on other repositories and has successfully worked for eXist-db documentation (https://github.com/eXist-db/documentation/blob/master/pom.xml). **But** I have not setup a test env to check if this will really work on the first try. Running the plugin in `-Dgithub.draft=true` works perfectly fine but no artifacts are uploaded which might be due to "draft mode". My expectation is that the plugin will work fine and not fail, in case it misses to upload a resource we can always manually upload it to the Github release or better, fix the configuration of the plugin.

### CODE REVIEW

**Comment by windauer** [comment]
to see if the release plugin really works as expected for eXist-db I could create a "eXist-db-test-release" by changing the configuration. That would be publicly visible till I delete it again but then we would know for sure that all files got uploaded and that the plugin does not run into problem because some artefacts are uploaded from exist-distribution and the other from exist-installer..

**Review by dizzzz — APPROVED** [review]

**Comment by line-o** [comment]
At the community call it was consensus that if the plugin itself works when called alone the we can safely expect it to work in the release process.

**Comment by adamretter** [comment]
> to see if the release plugin really works as expected for eXist-db I could create a "eXist-db-test-release" by changing the configuration

I would suggest to run the plugin standalone and check that it did upload your test artifacts to GitHub Releases, i.e. this will show that you have your `settings.xml` correct and the changes in your `pom.xml` correct. You can then remove the test artifacts from GitHub Releases.

As it stands, it looks to me like this won't work as something is missing from `settings.xml`, i.e. whilst you have the access token, the Github username appears to be missing. Is that not required?

**Comment by line-o** [comment]
It looks as if the upload was successful
https://exist-db.slack.com/archives/CG4AMEF0W/p1624372648025000

File: exist-distribution/pom.xml
Comment (adamretter) [inline exist-distribution/pom.xml:1093]: Syntax here and below is wrong you have a superfluous ')'

File: exist-installer/pom.xml
Comment (adamretter) [inline exist-installer/pom.xml:141]: Version should not be here if it is defined in the parent pom.xml

File: exist-versioning-release.md
Comment (adamretter) [inline exist-versioning-release.md:283]: I would not have thought you need to do this if you are using the GitHub Release plugin?

File: exist-versioning-release.md
Comment (adamretter) [inline exist-versioning-release.md:126]: Is it not worth mentioning that you need repo maintainer status on the eXist-db GitHub org (to replace the line about BinTray). I don't think just anyone can upload a release to GitHub, you need certain org/repo privileges I suspect.

**Review by adamretter — CHANGES_REQUESTED** [review]
Some small fixes required.

**Comment by adamretter** [comment]
> It looks as if the upload was successful

@line-o I am afraid that is a tag and not a release. Also I would avoid naming tags or releases with `-SNAPSHOT` suffix, I could imagine that could cause great pain with Maven's concept that a SNAPSHOT is a moving target.

**Comment by line-o** [comment]
Those were clearly just tests.

File: exist-versioning-release.md
Comment (adamretter) [inline exist-versioning-release.md:153]: What about the GitHub username? I don't think the token by itself is enough

**Comment by adamretter** [comment]
> Those were clearly just tests.

Sure! I wasn't disputing that they were tests...

File: exist-versioning-release.md
Comment (windauer) [reply exist-versioning-release.md:153]: a Github Userame is ont required but the <privateKey> is enough. Has been tested with several repos and works perfectly fine. Other way round, in case you have 2FactorAuth enabled (which we all have in the eXist-db github org) username / password will not work.

**Comment by windauer** [comment]
> > to see if the release plugin really works as expected for eXist-db I could create a "eXist-db-test-release" by changing the configuration
> 
> I would suggest to run the plugin standalone and check that it did upload your test artifacts to GitHub Releases, i.e. this will show that you have your `settings.xml` correct and the changes in your `pom.xml` correct. You can then remove the test artifacts from GitHub Releases.
> 
> As it stands, it looks to me like this won't work as something is missing from `settings.xml` (or your documentation of it in the .md file), i.e. whilst you have the access token, the Github username appears to be missing. Is that not required?

like I wrote in `Known Issues` in the PR description the Plugin has been tested standalone and works fine. But for obvious reasons it was only tested in draft mode where the connection is checked but uploads are not performed, so there is a slight risk here that something wont work. Thanks for your finding of ')' cause that would have certainly not worked!

**Comment by wolfgangmm** [comment]
It appears to me that all open questions were answered, so I would merge this if nobody shouts?

**Comment by wolfgangmm** [comment]
Ok, no further comments since 5 hours. I'll thus assume it's good to go and we give it a try.

**Pull request merged**
